contracts/tokenomics/generator/src/contract.rs

```diff
@@ -181,22 +185,30 @@ pub fn execute(
         ExecuteMsg::MassUpdatePools {} => {
             update_rewards_and_execute(deps, env, None, ExecuteOnReply::MassUpdatePools {})
         }
-        ExecuteMsg::UpdatePool { lp_token } => update_rewards_and_execute(
-            deps,
-            env,
-            Some(lp_token.clone()),
-            ExecuteOnReply::UpdatePool { lp_token },
-        ),
-        ExecuteMsg::Withdraw { lp_token, amount } => update_rewards_and_execute(
-            deps,
-            env,
-            Some(lp_token.clone()),
-            ExecuteOnReply::Withdraw {
-                lp_token,
-                account: info.sender.to_string(),
-                amount,
-            },
-        ),
+        ExecuteMsg::UpdatePool { lp_token } => {
+            let lp_token = addr_validate_to_lower(deps.api, &lp_token)?;
+
+            update_rewards_and_execute(
+                deps,
+                env,
+                Some(lp_token.to_string()),
```

Comment:
Let's use Addr type for update_single_pool parameter of update_rewards_and_execute function

Reply:
Ok, done